### PR TITLE
Update to Linux 5.10.179

### PR DESCRIPTION
**Pull request by saiarcot895**
Update from 5.10.140 to 5.10.179 (published as [linux 5.10.179-3](https://tracker.debian.org/news/1449040/accepted-linux-510179-3-source-into-oldstable-security/) by Debian). This brings in a number of bug and security fixes.

Update our ABI version number to 23-2 to match the ABI version 23 used by Debian for 5.10.179.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -6,7 +6,7 @@ KERNEL_ABI_MINOR_VERSION = 2
 KVERSION_SHORT ?= 5.10.0-23-$(KERNEL_ABI_MINOR_VERSION)
 KVERSION ?= $(KVERSION_SHORT)-amd64
 KERNEL_VERSION ?= 5.10.179
-KERNEL_SUBVERSION ?= 1
```
Comment (paulmenzel) [inline Makefile:9]: What is the subversion, and why change it?

File: patch/preconfig/packaging-update-abiname.patch [hunk 2]
```diff
@@ -9,26 +9,26 @@ From: Olivier Singla <olivier.singla@broadcom.com>
...
-index cd4938ecc..e54c10256 100644
+index 983625720..55d7e51a7 100644
 --- a/debian/config/defines
 +++ b/debian/config/defines
 @@ -1,5 +1,5 @@
  [abi]
--abiname: 18
-+abiname: 18-2
+-abiname: 23
++abiname: 23-2
```
Comment (paulmenzel) [inline patch/preconfig/packaging-update-abiname.patch:18]: It’d be great if you elaborated in the commit message, why 23.

**Review by paulmenzel — COMMENTED** [review]
Thank you for doing this update.

It’d be great if you mentioned the devices you tested this on.

I wouldn’t put a dot/period at the end of commit messages summaries.

File: Makefile (hunk 1 above)
Comment (saiarcot895) [reply Makefile:9]: The subversion refers to the Debian version number. In this case, version 5.10.179-3 from Debian is being used.

File: patch/preconfig/packaging-update-abiname.patch (hunk 2 above)
Comment (saiarcot895) [reply patch/preconfig/packaging-update-abiname.patch:18]: ABI 23 is the ABI version number that Debian is using. This typically gets incremented by one for each minor kernel update that Debian does (i.e. going from 5.10.179 to the current 5.10.191).

Comment (paulmenzel) [reply patch/preconfig/packaging-update-abiname.patch:18]: Indeed. Please reference the Debian package you base your change on. For example I found [different versions](https://tracker.debian.org/pkg/linux-signed-5.10-arm64): [version 5.10.179+3~deb10u1](https://tracker.debian.org/news/1449695/accepted-linux-signed-510-arm64-5101793deb10u1-source-into-oldoldstable/) and [5.10.179+5~deb10u1](https://tracker.debian.org/news/1452052/accepted-linux-signed-510-arm64-5101795deb10u1-source-into-oldoldstable/), where the change-log of the latter contains:

> Changes:
>  linux-signed-5.10-arm64 (5.10.179+5~deb10u1) buster-security; urgency=high
>  .
>    * Sign kernel from linux-5.10 5.10.179-5~deb10u1
>  .
>    * Rebuild for buster:
>      - Change ABI number to 0.deb10.24

Comment (saiarcot895) [reply patch/preconfig/packaging-update-abiname.patch:18]: I think the versions you found are for Buster (hinted by the `~deb10u1` at the end, which is used for backports).

I've updated the main PR description to include the version number and a link to the publishing of that version.

**Comment by saiarcot895** [comment]
FYI @keboliu @vivekrnv, several Mellanox patches have been disabled since they are present in 5.10.179.

**Review by paulmenzel — APPROVED** [review]
Feel free to address the nits I mentioned, but all in all, it looks great. Thank you for updating the description.

**Review by lguohan — APPROVED** [review]

**Comment by saiarcot895** [comment]
Backport ADO: part of 24592132

**Comment by StormLiangMS** [comment]
@saiarcot895 this PR failed compile with 202305, could you submit separate PR to 202305?

**Comment by saiarcot895** [comment]
Hi @StormLiangMS, for backporting, there would need to be additional separate changes on the sonic-buildimage repo side. I'll create manual PRs for both.

**Pull request merged**